src/compiler/checker.ts

```diff
@@ -14040,10 +14040,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return modifiers & MappedTypeModifiers.ExcludeOptional ? -1 : modifiers & MappedTypeModifiers.IncludeOptional ? 1 : 0;
     }
 
+    function getModifiersTypeOptionality(type: Type): number {

```

Comment:
I don't understand what this is computing. What does a result of `2` mean?

Reply:
The result is always either -1, 0, or 1. -1 means optionality is stripped (i.e. `-?`), 0 means optionality is unchanged, 1 means optionality is added (i.e. `+?`). When a homomorphic mapped type doesn't modify optionality, we (recursively) consult the optionality of the type you're mapping over to see if it strips or adds optionality.

Reply:
Should this be an enum? Then it'd be easier to tell what the numbers mean.

Reply:
It could be an enum, but `number` feels more appropriate since we use the greater and less than operators to relate the combined values.

Reply:
Would a comment about what each number represents work?